apps/system/lockscreen/js/lockscreen.js

```diff
@@ -335,125 +332,127 @@
-    window.addEventListener('iac-mediacomms', this);
-    window.addEventListener('appterminated', this);
-
-    // Listen to event to start the Camera app
-    window.addEventListener('holdcamera', this);
-
-    window.SettingsListener.observe('lockscreen.enabled', true,
-      (function(value) {
-        this.setEnabled(value);
-    }).bind(this));
-
-    // it is possible that lockscreen is initialized after wallpapermanager
-    // (e.g. user turns on lockscreen in settings after system is booted);
-    // if this is the case, then the wallpaperchange event might not be captured
-    //   and the lockscreen would initialize into empty wallpaper
-    // so we need to see if there is already a wallpaper blob available
-    if (Service.query('getWallpaper')) {
-      var wallpaperURL = Service.query('getWallpaper');
-      if (wallpaperURL) {
-        this.updateBackground(wallpaperURL);
-        this.overlay.classList.remove('uninit');
+      this.getAllElements();
+      this.notificationsContainer =
+        document.getElementById('notifications-lockscreen-container');
+
+      this.lockIfEnabled(true);
+      this.initUnlockerEvents();
+
+      // This component won't know when the it get locked unless
+      // it listens to this event.
+      window.addEventListener('lockscreen-appopened', this);
+
+      /* Status changes */
+      window.addEventListener(
+        'lockscreen-notification-request-activate-unlock', this);
+      window.addEventListener('screenchange', this);
+
+      /* Incoming and normal mode would be different */
+      window.addEventListener('lockscreen-mode-switch', this);
+
+      /* Gesture */
+      this.area.addEventListener('touchstart', this);
+      this.altCameraButton.addEventListener('click', this);
+      this.iconContainer.addEventListener('touchstart', this);
+
+      /* Unlock & camera panel clean up */
+      this.overlay.addEventListener('transitionend', this);
+
+      /* switching panels */
+      window.addEventListener('home', this);
+
+      /* blocking holdhome and prevent Cards View from show up */
+      window.addEventListener('holdhome', this, true);
+      window.addEventListener('ftudone', this);
+      window.addEventListener('moztimechange', this);
+      window.addEventListener('timeformatchange', this);
+
+      /* media playback widget */
+      this.mediaPlaybackWidget =
+        new window.LockScreenMediaPlaybackWidget(this.mediaContainer);
+
+      // listen to media playback events to adjust notification container height
+      window.addEventListener('iac-mediacomms', this);
+      window.addEventListener('appterminated', this);
+
+      // Listen to event to start the Camera app
+      window.addEventListener('holdcamera', this);
+
+      window.SettingsListener.observe('lockscreen.enabled', true,
+        (function(value) {
+          this.setEnabled(value);
+      }).bind(this));
+
+      // it is possible that lockscreen is initialized after wallpapermanager
```

Comment:
There is some overlap with https://github.com/mozilla-b2g/gaia/pull/31109 . One patch will need rebasing.